connect/transforms/src/main/java/org/apache/kafka/connect/transforms/field/SingleFieldPath.java

```diff
@@ -0,0 +1,500 @@
+                    if (atEndOfPath) {
+                        field.append(pathText, start, idx);
+                        // we've reached the end of the path, and the last character is the backtick
+                        steps.add(field.toString());
+                        idx++;
+                        break;
+                    }
+
+                    boolean notFollowedByDot = pathText.charAt(idx + 1) != DOT;
+                    if (notFollowedByDot) {
+                        boolean afterABackslash = pathText.charAt(idx - 1) == BACKSLASH;
+                        if (afterABackslash) {
+                            // this backtick was escaped; include it in the field name, but continue
+                            // looking for an unescaped matching backtick
+                            field.append(pathText, start, idx - 1)
+                                .append(BACKTICK);
+
+                            idx++;
+                            start = idx;
+                        } else {
+                            // this backtick isn't followed by a dot; include it in the field name, but continue
+                            // looking for a matching backtick that is followed by a dot
+                            idx++;
+                        }
+                        continue;
+                    }
+
+                    boolean afterABackslash = pathText.charAt(idx - 1) == BACKSLASH;
+                    if (afterABackslash) {
+                        // this backtick was escaped; include it in the field name, but continue
+                        // looking for an unescaped matching backtick
+                        field.append(pathText, start, idx - 1)
+                            .append(BACKTICK);
+
+                        idx++;
+                        start = idx;
+                        continue;
+                    }
+                    // we've found our matching backtick
+                    field.append(pathText, start, idx);
+                    steps.add(field.toString());
+                    idx += 2; // increment by two to include the backtick and the dot after it
+                    break;
+                }
+
+
+            }
+        }
+        return steps.toArray(new String[0]);
+    }
+
+
+    /**
+     * Access a {@code Field} at the current path within a schema {@code Schema}
+     * If field is not found, then {@code null} is returned.
+     */
+    public Field fieldFrom(Schema schema) {
+        if (path.length == 1) {
+            return schema.field(path[0]);
+        } else {
+            Schema current = schema;
+            for (int i = 0; i < path.length; i++) {
+                if (current == null) {
+                    return null;
+                }
+                if (i == path.length - 1) { // get value
+                    return current.field(path[i]);
+                } else { // iterate
+                    current = current.field(path[i]).schema();
```

Comment:
Won't this cause an NPE if the field doesn't exist in this schema?

Reply:
Yeap, good catch. Adding null validations.